### PR TITLE
🐛 fix for terraform <cannot get hcl block> err

### PR DESCRIPTION
**Pull request by vjeffrey**
fixes https://github.com/mondoohq/cnquery/issues/1737

### CODE REVIEW

File: providers/terraform/resources/hcl.go [hunk 1]
```diff
@@ -172,24 +172,16 @@ func newMqlHclBlock(runtime *plugin.Runtime, block *hcl.Block, file *hcl.File) (
...
-			r.MqlResource().Cache.Store("_hclblock", &resources.CacheEntry{
-				Data: block,
-			})
-			r.MqlResource().Cache.Store("_hclfile", &resources.CacheEntry{
-				Data: file,
-			})
-		}
-	*/
+	r := res.(*mqlTerraformBlock)
+	r.block = plugin.TValue[*hcl.Block]{State: plugin.StateIsSet, Data: block}
```
Comment (preslavgerchev) [inline providers/terraform/resources/hcl.go:186]: can't we pass this in the args as an mql resource, i.e. `llx.ResourceData`?

File: providers/terraform/resources/hcl.go [hunk 2]
```diff
@@ -172,24 +172,16 @@ func newMqlHclBlock(runtime *plugin.Runtime, block *hcl.Block, file *hcl.File) (
...
 
-	r, err := CreateResource(runtime, "terraform.block", map[string]*llx.RawData{
+	res, err := CreateResource(runtime, "terraform.block", map[string]*llx.RawData{
 		"type":    llx.StringData(block.Type),
 		"labels":  llx.ArrayData(llx.TArr2Raw(block.Labels), types.String),
 		"start":   llx.ResourceData(start, "terraform.fileposition"),
 		"end":     llx.ResourceData(end, "terraform.fileposition"),
 		"snippet": llx.StringData(snippet),
 	})
-
```
Comment (chris-rock) [inline providers/terraform/resources/hcl.go:182]: we need to err out when err != nil, otherwise we get a panic

**Review by chris-rock — CHANGES_REQUESTED** [review]
We need to add error handling, then LGTM

File: providers/terraform/resources/hcl.go (hunk 1 above)
Comment (vjeffrey) [reply providers/terraform/resources/hcl.go:186]: this is for the internal implementation, 

```

type mqlTerraformBlockInternal struct {
	block      plugin.TValue[*hcl.Block]
	cachedFile plugin.TValue[*hcl.File]
}
```

**Review by jaym — APPROVED** [review]

**Pull request merged**